src/lib.rs

```diff
@@ -2096,6 +2266,324 @@ where
+    /// OPTIONAL.  Human-readable identifier for the resource owner who
+    /// authorized this token.
+    ///
+    fn username(&self) -> Option<&String>;
+    ///
+    /// OPTIONAL.  Type of the token as defined in [Section 5.1](https://tools.ietf.org/html/rfc7662#section-5.1) of OAuth
+    /// 2.0 [RFC6749].
+    /// Value is case insensitive and deserialized to the generic `TokenType` parameter.
+    ///
+    fn token_type(&self) -> Option<&TT>;
+    ///
+    /// OPTIONAL.  Integer timestamp, measured in the number of seconds
+    /// since January 1 1970 UTC, indicating when this token will expire,
+    /// as defined in JWT [RFC7519](https://tools.ietf.org/html/rfc7519).
+    ///
+    fn exp(&self) -> Option<&DateTime<Utc>>;
+    ///
+    /// OPTIONAL.  Integer timestamp, measured in the number of seconds
+    /// since January 1 1970 UTC, indicating when this token was
+    /// originally issued, as defined in JWT [RFC7519](https://tools.ietf.org/html/rfc7519).
+    ///
+    fn iat(&self) -> Option<&DateTime<Utc>>;
+    ///
+    /// OPTIONAL.  Integer timestamp, measured in the number of seconds
+    /// since January 1 1970 UTC, indicating when this token is not to be
+    /// used before, as defined in JWT [RFC7519](https://tools.ietf.org/html/rfc7519).
+    ///
+    fn nbf(&self) -> Option<&DateTime<Utc>>;
+    ///
+    /// OPTIONAL.  Subject of the token, as defined in JWT [RFC7519](https://tools.ietf.org/html/rfc7519).
+    /// Usually a machine-readable identifier of the resource owner who
+    /// authorized this token.
+    ///
+    fn sub(&self) -> Option<&String>;
+    ///
+    /// OPTIONAL.  Service-specific string identifier or list of string
+    /// identifiers representing the intended audience for this token, as
+    /// defined in JWT [RFC7519](https://tools.ietf.org/html/rfc7519).
+    ///
+    fn aud(&self) -> Option<&Vec<String>>;
+    ///
+    /// OPTIONAL.  String representing the issuer of this token, as
+    /// defined in JWT [RFC7519](https://tools.ietf.org/html/rfc7519).
+    ///
+    fn iss(&self) -> Option<&String>;
+    ///
+    /// OPTIONAL.  String identifier for the token, as defined in JWT
+    /// [RFC7519](https://tools.ietf.org/html/rfc7519).
+    ///
+    fn jti(&self) -> Option<&String>;
+}
+
+///
+/// Standard OAuth2 token introspection response.
+///
+/// This struct includes the fields defined in
+/// [Section 2.2 of RFC 7662](https://tools.ietf.org/html/rfc7662#section-2.2), as well as
+/// extensions defined by the `EF` type parameter.
+///
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub struct StandardTokenInspectionResponse<EF, TT>
+where
+    EF: ExtraTokenFields,
+    TT: TokenType + 'static,
+{
+    active: bool,
+    #[serde(rename = "scope")]
+    #[serde(deserialize_with = "helpers::deserialize_space_delimited_vec")]
+    #[serde(serialize_with = "helpers::serialize_space_delimited_vec")]
+    #[serde(skip_serializing_if = "Option::is_none")]
+    #[serde(default = "none_field")]
```

Comment:
I'm pretty sure serde does this by default for all `Option` fields, so I think we can remove all of these `default = "none_field"` attributes

Reply:
I thought so too, but as far as I remember it did not work without the attribute.
I'll verify it when I update this PR.

Reply:
I verified, I was able to remove `#[serde(default = "none_field")]` from fields like `username: Option<String>` however, when the field has other attributes like `#[serde(deserialize_with` or `#[serde(with = ` the attribute seems to be required.
If you remove them the unit tests fail and complain about the field being missing.